### PR TITLE
gc: Update to 8.2.2

### PR DESCRIPTION
**Pull request by ivmai**
Topic Description
-----------------

Update libgc from 8.0.6 to 8.2.2

Package(s) Affected
-------------------

gc

Security Update?
----------------

No

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
    

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

### CODE REVIEW

**Comment by ivmai** [comment]
@MingcongBai or @liushuyu, could you please review?

**Comment by MingcongBai** [comment]
Code linting pending, I will take over this branch.

**Comment by ivmai** [comment]
@MingcongBai Please take care of. I'm getting build failure reports daily(

**Comment by ivmai** [comment]
@CamberLoid Could you please review this PR?

**Comment by CamberLoid** [comment]
@ivmai Noticed. I will see what I can do.

**Comment by CamberLoid** [comment]
@MingcongBai Request a review. 
No major sonames version bump, so no rebuilding is required.

**Comment by MingcongBai** [comment]
Approving. Please help building these packages.

File: base-libs/gc/autobuild/beyond [hunk 1]
```diff
@@ -1,3 +1,4 @@
-sed -i -e 's/GC_MALLOC 3/gc 3/' doc/gc.man
-install -m755 -d "$PKGDIR/usr/share/man/man3"
-install -m644 doc/gc.man "$PKGDIR/usr/share/man/man3/gc.3"
+abinfo "Modifying and installing manual pages ..."
+sed -i -e 's/BDWGC 3/gc 3/' doc/gc.man
```
Comment (MingcongBai) [inline base-libs/gc/autobuild/beyond]: Please use absolute paths - `sed -i -e 's/BDWGC 3/gc 3/' "$SRCDIR"/doc/gc.man`. Ditto for line 4.

File: base-libs/gc/autobuild/patch [hunk 2]
```diff
@@ -1,6 +1,4 @@
+abinfo "Hacking docs makefile ..."
 sed -i 's#pkgdata#doc#' doc/doc.am
-for i in autobuild/patches/*.patch
-do
-	patch -Np1 -i $i
-done
-autoreconf -fi
+
+# autoreconf -fi
```
Comment (MingcongBai) [inline base-libs/gc/autobuild/patch]: Remove all unneeded lines, don't comment.

File: base-libs/gc/autobuild/patch [hunk 3]
```diff
@@ -1,6 +1,4 @@
+abinfo "Hacking docs makefile ..."
 sed -i 's#pkgdata#doc#' doc/doc.am
```
Comment (MingcongBai) [inline base-libs/gc/autobuild/patch]: Please use absolute path, see above.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Comment by MingcongBai** [comment]
@ivmai Apologies, but there are a few places that needs changes. Could you please make some revisions?

File: base-libs/gc/autobuild/beyond (hunk 1 above)
Comment (ivmai) [reply base-libs/gc/autobuild/beyond]: done

File: base-libs/gc/autobuild/patch (hunk 3 above)
Comment (ivmai) [reply base-libs/gc/autobuild/patch]: done

**Comment by ivmai** [comment]
> Could you please make some revisions?

Done

**Comment by CamberLoid** [comment]
Unit tests are passed on amd64 platform. 
image
I will try building & testing on riscv64 and loongson3/mips64le platform. Will approve if no further issue found.

**Comment by CamberLoid** [comment]
LGTM on riscv64 platform.

**Comment by CamberLoid** [comment]
LGTM on loongson3 platform.

**Review by CamberLoid — APPROVED** [review]

**Review by MingcongBai — APPROVED** [review]

**Comment by MingcongBai** [comment]
> > Could you please make some revisions?
> 
> Done

Thank you, approved.

**Pull request merged**